base/bucket_gocb.go

```diff
@@ -952,8 +896,85 @@ func (bucket CouchbaseBucketGoCB) Update(k string, exp int, callback sgbucket.Up
 }
 
 func (bucket CouchbaseBucketGoCB) WriteUpdate(k string, exp int, callback sgbucket.WriteUpdateFunc) error {
-	LogPanic("Unimplemented method: WriteUpdate()")
-	return nil
+
+	bucket.singleOps <- struct{}{}
+	defer func() {
+		<-bucket.singleOps
+	}()
+
+	// The number of Couchbase Nodes that durable writes should be replicated to or persisted to
+	numNodesReplicateTo := uint(0)
+	numNodesPersistTo := uint(1)
+
+	maxCasRetries := 100000 // prevent infinite loop
```

Comment:
Agreed, this feels kludgy and confusing.  I changed to a plain `for {..}` until we can come up with a better way